core/file.go

```diff
@@ -75,16 +75,43 @@ func (file *File) State() (llb.State, error) {
 	return defToState(file.LLB)
 }
 
+// Contents handles file content retrieval
 func (file *File) Contents(ctx context.Context, gw bkgw.Client) ([]byte, error) {
 	return WithServices(ctx, gw, file.Services, func() ([]byte, error) {
 		ref, err := gwRef(ctx, gw, file.LLB)
 		if err != nil {
 			return nil, err
 		}
 
-		return ref.ReadFile(ctx, bkgw.ReadRequest{
-			Filename: file.File,
-		})
+		// Stat the file and preallocate file contents buffer:
+		st, err := file.Stat(ctx, gw)
+		if err != nil {
+			return nil, err
+		}
+
+		fileSize := int(st.GetSize_())
+		contents := make([]byte, fileSize)
```

Comment:
I think we should probably keep some limit on the max size we will read, just so we stop users from accidentally trying to read an e.g. 100GB file into memory.

For now, I'd be okay with just having a const for this max. We could then consider making it a parameter in the future if anyone requests it, but not needed now.

In terms of the value, my pretty much arbitrary made up number would be 128MB :-) That seems sufficiently high to cover most cases without getting too unreasonable, but open to suggestions